percona_tests/metrics_test.go

```diff
@@ -0,0 +1,512 @@
+	}
+
+	return true, ""
+}
+
+func testForMissingMetrics(oldMetricsCollection, newMetricsCollection MetricsCollection) (bool, string) {
+	missingMetrics := make([]string, 0)
+	for metricName := range oldMetricsCollection.LabelsByMetric {
+		if _, ok := newMetricsCollection.LabelsByMetric[metricName]; !ok {
+			missingMetrics = append(missingMetrics, metricName)
+		}
+	}
+	sort.Strings(missingMetrics)
+	if len(missingMetrics) > 0 {
+		return false, fmt.Sprintf("Missing metrics (%d items):\n%s", len(missingMetrics), strings.Join(missingMetrics, "\n"))
+	}
+
+	return true, ""
+}
+
+func dumpMetricsWithMultipleLabelSets(newMetricsCollection MetricsCollection) {
+	metricsWithMultipleLabels := make(map[string][]string)
+	for metricName, newMetricLabels := range newMetricsCollection.LabelsByMetric {
+		if len(newMetricLabels) > 1 {
+			found := false
+			for i := 0; !found && i < len(newMetricLabels); i++ {
+				lbl := newMetricLabels[i]
+
+				for j := 0; j < len(newMetricLabels); j++ {
+					if i == j {
+						continue
+					}
+
+					lbl1 := newMetricLabels[j]
+					if lbl == "" || lbl1 == "" {
+						continue
+					}
+
+					if strings.Contains(lbl, lbl1) || strings.Contains(lbl1, lbl) {
+						found = true
+						break
+					}
+				}
+			}
+			if found {
+				metricsWithMultipleLabels[metricName] = newMetricLabels
+			}
+		}
+	}
+
+	if len(metricsWithMultipleLabels) > 0 {
+		ss := make([]string, 0, len(metricsWithMultipleLabels))
+		for k, v := range metricsWithMultipleLabels {
+			ss = append(ss, fmt.Sprintf("%s\n        %s", k, strings.Join(v, "\n        ")))
+		}
+		fmt.Printf("Some metrics were collected multiple times with extra labels (%d items):\n    %s\n\n", len(metricsWithMultipleLabels), strings.Join(ss, "\n    "))
+	}
+}
+
+func dumpExtraMetrics(newMetricsCollection, oldMetricsCollection MetricsCollection) {
+	extraMetrics := make([]string, 0)
+	for metricName := range newMetricsCollection.LabelsByMetric {
+		if _, ok := oldMetricsCollection.LabelsByMetric[metricName]; !ok {
+			extraMetrics = append(extraMetrics, metricName)
+		}
+	}
+	sort.Strings(extraMetrics)
+
+	if len(extraMetrics) > 0 {
+		fmt.Printf("Extra metrics (%d items):\n    %s\n\n", len(extraMetrics), strings.Join(extraMetrics, "\n    "))
+	}
+}
+
+func parseMetricsCollection(metricRaw string) MetricsCollection {
+	rawMetricsArr := strings.Split(metricRaw, "\n")
+	metricNamesArr := getMetricNames(rawMetricsArr)
+	metrics := parseMetrics(metricNamesArr)
+	labelsByMetrics := groupByMetrics(metrics)
+
+	return MetricsCollection{
+		MetricNamesWithLabels: metricNamesArr,
+		MetricsData:           metrics,
+		RawMetricStr:          metricRaw,
+		RawMetricStrArr:       rawMetricsArr,
+		LabelsByMetric:        labelsByMetrics,
+	}
+}
+
+func arrIsSubsetOf(a, b []string) bool {
+	if len(a) == 0 {
+		return len(b) == 0
+	}
+
+	for _, x := range a {
+		if !contains(b, x) {
+			return false
+		}
+	}
```

Comment:
I would implement it with `set`(`map[string]struct{}`), in that case, complexity will be `N` instead of `N^2`